src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -26,6 +26,7 @@ public sealed class TelemetryConfiguration : IDisposable
         private string instrumentationKey = string.Empty;
         private bool disableTelemetry = false;
         private TelemetryProcessorChainBuilder builder;
+        private SnapshottingList<IMetricProcessor> metricProcessors = new SnapshottingList<IMetricProcessor>();
```

Comment:
I'm reusing what's already available. I do not like it very much either but do not intend to fix that particular problem in this effort.